internal/serve/serve.go

```diff
@@ -150,10 +150,33 @@ func (opts *ServeOptions) SetupDependencies() error {
 	return nil
 }
 
+// ValidateSecurity validates the MFA and ReCAPTCHA security options.
+func (opts *ServeOptions) ValidateSecurity() error {
+	if opts.NetworkPassphrase == network.PublicNetworkPassphrase {
+		if opts.DisableMFA {
+			return fmt.Errorf("MFA cannot be disabled in pubnet")
+		} else if opts.DisableReCAPTCHA {
+			return fmt.Errorf("reCAPTCHA cannot be disabled in pubnet")
+		}
+	}
+
+	if opts.DisableMFA {
+		log.Warnf("MFA is disabled in network '%s'", opts.NetworkPassphrase)
```

Comment:
nit: we can get the network to not log the passphrase. But really optional change, especially that the network here is always going to be testnet (we error if it's disabled and it's pubnet)
```go
networkType, err := utils.GetNetworkTypeFromNetworkPassphrase(opts.NetworkPassphrase)
```

Reply:
I decided to keep it as is and not use GetNetworkTypeFromNetworkPassphrase for now, also because it's not being widely used in the project...

We could do a pass and apply it everywhere later. Plz let me know if you disagree.